main.go

```diff
@@ -0,0 +1,258 @@
+	cmdComponents := strings.Split(cmd, " ")
+
+	return &Exporter{
+		cmd:  cmdComponents[0],
+		args: cmdComponents[1:],
+		up: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "up"),
+			"Could passenger status be queried.",
+			nil,
+			nil,
+		),
+		uptime: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "uptime"),
+			"Number of seconds since passenger started.",
+			nil,
+			nil,
+		),
+		version: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "version"),
+			"Version of passenger",
+			[]string{"version"},
+			nil,
+		),
+		toplevelQueue: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "top_level_queue"),
+			"Number of requests in the top-level queue.",
+			nil,
+			nil,
+		),
+		maxProcessCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "max_process_count"),
+			"Configured maximum number of processes.",
+			nil,
+			nil,
+		),
+		currentProcessCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "current_process_count"),
+			"Current number of processes.",
+			nil,
+			nil,
+		),
+		appCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "app_count"),
+			"Number of apps.",
+			nil,
+			nil,
+		),
+		appQueue: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "app_queue"),
+			"Number of requests in app process queues.",
+			[]string{"name"},
+			nil,
+		),
+		appProcsSpawning: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "app_procs_spawning"),
+			"Number of processes spawning.",
+			[]string{"name"},
+			nil,
+		),
+		requestsProcessed: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "requests_processed"),
+			"Number of processes served by a process.",
+			[]string{"name", "pid"},
+			nil,
+		),
+		procUptime: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "proc_uptime"),
+			"Number of seconds since processor started.",
+			[]string{"name", "pid"},
+			nil,
+		),
+		procMemory: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "proc_memory"),
+			"Memory consumed by a process",
+			[]string{"name", "pid"},
+			nil,
+		),
+		procStatus: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "proc_status"),
+			"Running status for a process.",
+			[]string{"name", "pid", "codeRevision", "lifeStatus", "enabled"},
+			nil,
+		),
+	}
+}
+
+// Collect fetches the statistics from the configured memcached server, and
+// delivers them as Prometheus metrics. It implements prometheus.Collector.
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	info, err := e.status()
+	if err != nil {
+		ch <- prometheus.MustNewConstMetric(e.up, prometheus.GaugeValue, 0)
+		log.Errorf("failed to collect status from passenger: %s", err)
+		return
+	}
+	ch <- prometheus.MustNewConstMetric(e.up, prometheus.GaugeValue, 1)
+
+	// TODO: Doesn't seem to be a queryable uptime on the main process.
+	// Talk with @grobie about how to handle this.
```

Comment:
Do you mean something like how long the passenger process is running already?

You could have a look at the process collector in the golang client. The memcached and haproxy exporters use that to additionally monitor another process as well. Have a look at their main() functions.